hipercow-api-unit-tests/Controllers/ClustersControllerUnitTest.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Imperial College London. All rights reserved.
+
+namespace Hipercow_api_unit_tests.Controllers
+{
+    using Hipercow_api.Controllers;
+    using Hipercow_api.Models;
+    using Hipercow_api.Tools;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// Test the /clusters endpoint.
+    /// </summary>
+    public class ClustersControllerUnitTest
+    {
+        /// <summary>
+        /// Test that /clusters tells us about wpia-hn.
+        /// </summary>
+        [Fact]
+        public void GetClusterCall_Works()
+        {
+            ClustersController cc = new ClustersController();
+            List<string> clusters = cc.Get();
+            Assert.Single(clusters);
+            Assert.Equal("wpia-hn", clusters.First());
+        }
+
+        /// <summary>
+        /// Test that /clusters/wrong handles incorrect
+        /// cluster name tidily.
+        /// </summary>
+        [Fact]
+        public void GetWrongCluster_ReturnsNotFound()
+        {
+            ClustersController cc = new ClustersController();
+            IActionResult res = cc.Get("turnip");
+            Assert.Equivalent(cc.NotFound(), res);
+        }
+
+        /// <summary>
+        /// Test just the endpoint handling of /clusters/cluster
+        /// by providing fake data.
+        /// </summary>
+        [Fact]
+        public void GetClusterinfo_Works()
+        {
+            ClustersController cc = new ClustersController();
+            FakeClusterInfoQuery fake = new FakeClusterInfoQuery();
```

Comment:
General comment that also appolies here.. probably use actual mocks for these (using Moq) can control these a lot better and avoid having to create fake classes... you will be able to do it like 
```
Mock<IClusterQuery> mockClusterQuery = new() 
mockClusterQuery.Setup(x => x.GetClusterInfo(It.IsAny<String>())).Returns(new ClusterInfo())
```

Reply:
Think this is done now in #3